statistics.cpp

```diff
@@ -52,7 +54,7 @@ double Statistics::log_likelihood(const VectorType& data,
     {
       if (distribution[i] < 0)
       {
-        pmf[i] = 1e-9;
+        pmf[i] = 1e-9 * max_conc / norm;
```

Comment:
Yes, this looks better to me.

The result here will not be normalized if you happen to have many 1e-9 values. It probably makes no real difference, though.

Reply:
I thought it didn't make a difference. I could take a pass through the vector first and find the maximum and then replace the substitution in the norm calculation.

Reply:
I agree that it's not likely going to make a difference.  I'll leave it to you.